anchore_engine/clients/grype_wrapper.py

```diff
@@ -231,32 +237,111 @@ def update_grype_db(grype_db_archive_local_file_location: str, version_name: str
             write_lock.release()
 
 
-def get_vulnerabilities(grype_sbom: str) -> json:
+def _get_grype_db_dir():
+    global grype_db_dir
+    return grype_db_dir
```

Comment:
I'm perhaps not understanding this well, but I'm not sure why `grype_db_dir` needs to be a global. Using a function here is definitely an improvement, but I can't reason why `global` is still being required

Reply:
Possibly I am not understanding the `global` correctly - my intention here (and previously when these variables were accessed elsewhere) was that when accessing these we always want the module-scoped `grype_db_dir` and `grype_db_session` declared above. This module differs from the other wrappers in that it needs to maintain state to be applied on all get/query vulnerabilities requests, and allow that state to be updated externally.

Coming from a Java background I also find allowing module-external, direct variable access to be not great, and would rather refactor this whole thing into a class with a method that returns a singleton instance of the class. But 1) I'd need to discuss the startup/instantiation implications of that with the rest of our feature group first, 2) that will break the pattern of the other wrappers (but may be worth it), and 3) that is really outside the scope of this change.

Reply:
I understand the concerns, but I'll have to push back on the reasoning that this is out of scope. This PR is introducing `globals()`, and I'm not finding a reasonable explanation to why that is a requirement. 
I'm not implying that a complete new mechanism needs to exist to handle singletons or configuration values. There are many different ways to declare variables, keep state, and let that state be shared on all consumers _besides_ using `global()`.

This could be:

1. a top-level variable in a module
2. a read-only implementation of a dictionary (or other data structure), to use as a variable
3. a value that gets passed in from function to function
4. an environment variable

I would gladly welcome any code that breaks away from current production code that implements anti-patterns like `globals()` instead of any other option - including singletons, module-level vars, or any other mechanism similar to what I've suggested. 

This  type of addition is going to be a -1 for me.